engine/src/main/java/org/hibernate/search/spi/impl/DelegatingIndexedTypeMap.java

```diff
@@ -29,6 +29,7 @@
 
 	private final Map<IndexedTypeIdentifier,V> map;
 	private final Map<String,IndexedTypeIdentifier> name2keyMapping;
+	private volatile IndexedTypeSet keyset;
```

Comment:
I don't understand, how is this related to removing the "put" method? Seems it isn't used, either.

Reply:
Indeed, it's part of a WIP I separated. Forgot to remove this piece.

FYI the (unrelated) work is that we use `keySet` quite regularly and it's not efficient to rebuild the returned `IndexedTypeSet` each time. I don't like the volatile path though, so more changes will be needed to fix that.

thanks for catching that!